keras/engine/topology.py

```diff
@@ -2746,6 +2746,28 @@ def preprocess_weights_for_loading(layer, weights,
     # Returns
         A list of weights values (Numpy arrays).
     """
+
+    if layer.__class__.__name__ == 'Bidirectional':
+        num_weights_per_layer = len(weights) // 2
+
+        forward_weights = _preprocess_weights(layer.forward_layer, weights[:num_weights_per_layer],
+                                              original_keras_version,
+                                              original_backend)
+        backward_weights = _preprocess_weights(layer.backward_layer, weights[num_weights_per_layer:],
+                                               original_keras_version,
+                                               original_backend)
+        weights = forward_weights + backward_weights
+    elif layer.__class__.__name__ == 'TimeDistributed':
+        layer = layer.layer
+        weights = _preprocess_weights(layer, weights, original_keras_version, original_backend)
+    else:
+        weights = _preprocess_weights(layer, weights, original_keras_version, original_backend)
+
+    return weights
+
+
+def _preprocess_weights(layer, weights, original_keras_version=None, original_backend=None):
```

Comment:
I felt like it made more sense to do separate calls for `layer.forward_layer` and `layer.backward_layer` in the bidirectional case since then you are passing an actual instance of the correct child layer into `_preprocess_weights`. You could do it in a single method but it would involve either duplicating code or specifically tailoring parts of `_preprocess_weights` to  cater for the bidirectional case where there are twice as many weight tensors, and the list of weight tensors is hard-coded with indices in that method already.